### DISCUSSION TITLE
suggestion: Add support for `FontCollection::getFontManagerOrder`

### DISCUSSION THREAD

**Discussion by marc2332** [rust-skia/rust-skia#930] [Polls]
hey Armin!

Having [`FontCollection::getFontManagerOrder`](https://github.com/google/skia/blob/326bdc97ac40be329f672c263ae8a65eec362624/modules/skparagraph/src/FontCollection.cpp#L70) would be quite useful considering that [`svg::Dom::from_bytes`](https://github.com/rust-skia/rust-skia/commit/a86d196e687230f3d69175ee5016af9489eeb607#diff-65d64d273edf52c5b34b3b9acafc6ace6e83074d2b83bc2d18dae423e54245cbR184) now requires passing a `FontMgr`.

I think it should be easy to implement, right?  It would make things a bit easier, as I mainly work with `FontCollection` in Freya. I could try making a PR if you are busy, although I haven't tried compiling Skia in a while.

Thoughts ? 👀 😄

**Comment by pragmatrix** [comment]
Hey Marc,

the function itself is private, and it should be simple to get the functionality implemented on the Rust side, after all, Freya is providing all the font managers and the correct order should be resolvable when looking at the source. So although convenient, I don't think this should be part of skia-safe. 

There are a number of suggestion for more convenient or idiomatic Rust additions, which I personally really like to have (for example #888) and I for myself try to add rudimentary helpers so that the API does not feel completely alien, but thinking this through, and as long there is no clear definition how far this should go, I prefer to add only the absolute necessary ones to remove the maintenance cost and duplication.

I am open to hosting another crate that is based on skia-safe under rust-skia that implements such functionality, but only if another maintainer is taking care of it.

**Reply by Caellian** [reply]
Suggestions like #888 would be much simpler to maintain with layout tests that notified you of breakages when you rebase.

That suggestion _could_ be implemented externally, but having trait wrappers would feel alien in a lot of cases. Note that the example `to_vec` implementation would idiomatically be `Into<Vec<f32>>`.

There's also traits that can't be implemented for external types (`Deref`, `AsRef`, ...) which would then require creating wrappers that are a) a lot more work to maintain, b) don't integrate nicely into the API unless the crate wraps _all_ functions, c) end up being confusing for dependents or must be properly documented.

That's why I think something like layout tests would be much better, though they would still require you to update offsets when they _do_ break. I don't expect that to be that common however.
